src/Core/Resolvers/Factories/QueryManagerFactory.cs

```diff
@@ -0,0 +1,103 @@
+    {
+        private readonly IDictionary<DatabaseType, IQueryBuilder> _queryBuilders;
+        private readonly IDictionary<DatabaseType, IQueryExecutor> _queryExecutors;
+        private readonly IDictionary<DatabaseType, DbExceptionParser> _dbExceptionsParsers;
+
+        /// <summary>
+        /// Initiates an instance of QueryManagerFactory
+        /// </summary>
+        /// <param name="runtimeConfigProvider">runtimeconfigprovider.</param>
+        /// <param name="logger">logger.</param>
+        /// <param name="contextAccessor">httpcontextaccessor.</param>
+        public QueryManagerFactory(RuntimeConfigProvider runtimeConfigProvider, ILogger<IQueryExecutor> logger, IHttpContextAccessor contextAccessor)
+        {
+            _queryBuilders = new Dictionary<DatabaseType, IQueryBuilder>();
+            _queryExecutors = new Dictionary<DatabaseType, IQueryExecutor>();
+            _dbExceptionsParsers = new Dictionary<DatabaseType, DbExceptionParser>();
+            foreach (DataSource dataSource in runtimeConfigProvider.GetConfig().ListAllDataSources())
+            {
+                IQueryBuilder? queryBuilder = null;
+                IQueryExecutor? queryExecutor = null;
+                DbExceptionParser? exceptionParser = null;
+
+                switch (dataSource.DatabaseType)
+                {
+                    case DatabaseType.CosmosDB_NoSQL:
+                        break;
+                    case DatabaseType.MSSQL:
+                        queryBuilder = new MsSqlQueryBuilder();
+                        exceptionParser = new MsSqlDbExceptionParser(runtimeConfigProvider);
+                        queryExecutor = new MsSqlQueryExecutor(runtimeConfigProvider, exceptionParser, logger, contextAccessor);
+                        break;
+                    case DatabaseType.MySQL:
+                        queryBuilder = new MySqlQueryBuilder();
+                        exceptionParser = new MySqlDbExceptionParser(runtimeConfigProvider);
+                        queryExecutor = new MySqlQueryExecutor(runtimeConfigProvider, exceptionParser, logger, contextAccessor);
+                        break;
+                    case DatabaseType.PostgreSQL:
+                        queryBuilder = new PostgresQueryBuilder();
+                        exceptionParser = new PostgreSqlDbExceptionParser(runtimeConfigProvider);
+                        queryExecutor = new PostgreSqlQueryExecutor(runtimeConfigProvider, exceptionParser, logger, contextAccessor);
+                        break;
+                    default:
+                        throw new NotSupportedException(dataSource.DatabaseTypeNotSupportedMessage);
+                }
+
+                _queryBuilders.TryAdd(dataSource.DatabaseType, queryBuilder!);
+                _queryExecutors.TryAdd(dataSource.DatabaseType, queryExecutor!);
+                _dbExceptionsParsers.TryAdd(dataSource.DatabaseType, exceptionParser!);
+            }
```

Comment:
Let's say you have > 1 MsSQL, MySQL, PostgreSQL database sources. Wont' this code block create, for example of MsSql:
- > 1 instance of MsSqlQueryBuilder, MsSqlQueryExecutor, MsSqlDbExceptionParser, and then attempt to add them to the relevant dictionaries every time the new instances are created?
- I understand that you are using `TryAdd()` so the first instance wins, but that's a lot of new objects created that may not be used. Perhaps I missed something?

Reply:
This is correct. fixed this to create objects only once.